### PR TITLE
Fix SpinBox not clearing text on improper input 

### LINKED ISSUE #81071: SpinBox does not clear text on Enter key if the text is "Nil".
### Godot version

4.2.dev3.official

### System information

Windows 10 - GLES3 (Compatibility)

### Issue description

When a text is typed into the SpinBox node, and the `Enter` key is pressed, the text is normally removed, and the last valid number value of the SpinBox is restored.
However, when the string that is input is `"Nil"`, pressing `Enter` won't remove it, as it would do with any other text.

https://github.com/godotengine/godot/assets/141344817/a866a597-02db-4195-b45e-87578e0f1f92

### Steps to reproduce

0. Create a new project.
1. Add SpinBox node.
2. Run the project.
3. Input any valid number into the SpinBox.
4. Try to input some text into the SpinBox.
5. Try to type and input "Nil" text into the SpinBox.

### Minimal reproduction project

[SpinBox Nil - MRP.zip](https://github.com/godotengine/godot/files/12453969/SpinBox.Nil.-.MRP.zip)

### PR DESCRIPTION
**Pull request by OXTyler**
fixes https://github.com/godotengine/godot/issues/81071
The issue was when parsing errors are thrown, it does not revert back to original text

### CODE REVIEW

File: scene/gui/spin_box.cpp [hunk 1]
```diff
@@ -392,4 +393,4 @@ SpinBox::SpinBox() {
 	range_click_timer = memnew(Timer);
 	range_click_timer->connect("timeout", callable_mp(this, &SpinBox::_range_click_timeout));
 	add_child(range_click_timer, false, INTERNAL_MODE_FRONT);
-}
+}
```
Comment (AThousandShips) [inline scene/gui/spin_box.cpp:396]:
```suggestion
}

```
Missing line termination, though I don't know why you edited this line like this manually

Comment (OXTyler) [reply scene/gui/spin_box.cpp:396]: I accidentally deleted the file contents because I had the wrong window focused and had to copy paste it back, my bad

**Comment by AcatXIo** [comment]
Thanks for providing the fix!

**Review by KoBeWi — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**